presentation/components/src/presentation-components/properties/NavigationPropertyTargetSelector.tsx

```diff
@@ -0,0 +1,127 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import classnames from "classnames";
+import * as React from "react";
+import {
+  components, ControlProps, IndicatorContainerProps, IndicatorProps, MenuProps, OptionProps, OptionTypeBase, ValueContainerProps,
+} from "react-select";
+import { AsyncPaginate } from "react-select-async-paginate";
+import { PropertyRecord, PropertyValue, PropertyValueFormat } from "@itwin/appui-abstract";
+import { PropertyEditorProps } from "@itwin/components-react";
+import { IModelConnection } from "@itwin/core-frontend";
+import { SvgCaretDownSmall } from "@itwin/itwinui-icons-react";
+import { NavigationPropertyInfo } from "@itwin/presentation-common";
+import { translate } from "../common/Utils";
+import {
+  NavigationPropertyTarget, useNavigationPropertyTargetsLoader, useNavigationPropertyTargetsRuleset,
+} from "./UseNavigationPropertyTargetsLoader";
+
+export interface NavigationPropertyTargetSelectorAttributes {
+  getValue: () => PropertyValue | undefined;
+  divElement: HTMLDivElement | null;
+}
+
+export interface NavigationPropertyTargetSelectorProps extends PropertyEditorProps {
+  imodel: IModelConnection;
+  getNavigationPropertyInfo: (record: PropertyRecord) => Promise<NavigationPropertyInfo | undefined>;
+}
+
+export const NavigationPropertyTargetSelector = React.forwardRef<NavigationPropertyTargetSelectorAttributes, NavigationPropertyTargetSelectorProps>((props, ref) => {
+  const { imodel, getNavigationPropertyInfo, propertyRecord, onCommit, setFocus } = props;
+  const divRef = React.useRef<HTMLDivElement>(null);
+  const targetsRuleset = useNavigationPropertyTargetsRuleset(getNavigationPropertyInfo, propertyRecord);
+  const loadTargets = useNavigationPropertyTargetsLoader({ imodel, ruleset: targetsRuleset });
```

Comment:
We won't get the ruleset if `getNavigationPropertyInfo` is `undefined`, which means we also won't get any load targets. I suppose that means we're going to see an empty select box?

I think we should render a read-only record if it has a value.

Reply:
Changed to use default property renderer to render property value if `targetsRuleset` is `undefined`.